src/ai/ai_gui.cpp

```diff
@@ -124,15 +141,29 @@ struct AIListWindow : public Window {
 			case WID_AIL_LIST: {
 				/* Draw a list of all available AIs. */
 				int y = this->GetWidget<NWidgetBase>(WID_AIL_LIST)->pos_y;
+				Dimension rai = GetSpriteSize(SPR_AICONFIG_RANDOM);
+				uint rai_y_offset = (line_height - rai.height) / 2;
+				bool rtl = _current_text_dir == TD_RTL;
+				uint rai_left = rtl ? r.right - WD_MATRIX_RIGHT - rai.width : r.left + WD_MATRIX_LEFT;
 				/* First AI in the list is hardcoded to random */
 				if (this->vscroll->IsVisible(0)) {
-					DrawString(r.left + WD_MATRIX_LEFT, r.right - WD_MATRIX_LEFT, y + WD_MATRIX_TOP, this->slot == OWNER_DEITY ? STR_AI_CONFIG_NONE : STR_AI_CONFIG_RANDOM_AI, this->selected == -1 ? TC_WHITE : TC_ORANGE);
+					StringID text = STR_AI_CONFIG_RANDOM_AI;
+					if (this->slot == OWNER_DEITY) text = STR_AI_CONFIG_NONE;
+					uint text_width = GetStringBoundingBox(text).width;
+					uint text_left = rtl ? rai_left - 10 - text_width : rai_left + rai.width + 10;
+					uint text_right = rtl ? text_left + text_width : text_left + text_width;
+					DrawString(text_left, text_right, y + WD_MATRIX_TOP, text, this->selected == -1 ? TC_WHITE : TC_ORANGE);
```

Comment:
What is going on here? Why do you care about the text width?

Reply:
Look at how other code does this. You need the width of the images, not the text!